airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/thread_pool_manager.py

```diff
@@ -35,19 +34,10 @@ def __init__(
         self._futures: List[Future[Any]] = []
 
     def submit(self, function: Callable[..., Any], *args: Any) -> None:
-        # Submit a task to the threadpool, waiting if there are too many pending tasks
-        self._wait_while_too_many_pending_futures(self._futures)
+        # Submit a task to the threadpool, removing completed tasks if there are too many tasks in self._futures.
+        self._prune_futures(self._futures)
         self._futures.append(self._threadpool.submit(function, *args))
```

Comment:
How do we ensure the limit of futures if we don't wait for threads/futures to be completed? Is `self._threadpool` handling this under the hood?